workflows/rules/mapping/mapping.smk

```diff
@@ -26,79 +20,91 @@ def get_mapping(wcs):
-        temp("{workingdir}/{stype}/mapping/{fastqpattern}.bam")
+        bam = temp("{stype}/mapping/{fastqpattern}.bam"),
+        bai = temp("{stype}/mapping/{fastqpattern}.bam.bai")
     shell:
         "echo $HOSTNAME;"
         "{params.sentieon} bwa mem "
             "-M -R '@RG\\tID:{wildcards.fastqpattern}\\tSM:{params.samplename}\\tPL:ILLUMINA' "
-            "-t {params.threads} {params.referencegenome} {params.fwd_fmt} {params.rev_fmt} "
-        "| {params.sentieon} util sort -o {output} -t {params.threads} --sam2bam -i -"
+            "-t {params.threads} {params.referencegenome} {input.fwd_fmt} {input.rev_fmt} "
+        "| {params.sentieon} util sort -o {output.bam} -t {params.threads} --sam2bam -i -"
 
 rule dedup:
     input:
-        bamfiles = get_mapping
+        unpack(get_mapping)
     output:
-        temp("{workingdir}/{stype}/dedup/{sname}_DEDUP.bam"),
-        "{workingdir}/{stype}/dedup/{sname}_DEDUP.txt"
+        bam = temp("{stype}/dedup/{sname}_DEDUP.bam"),
+        bai = temp("{stype}/dedup/{sname}_DEDUP.bam.bai"),
+        score = temp("{stype}/dedup/{sname}_DEDUP_score.txt"),
+        metrics = temp("{stype}/dedup/{sname}_DEDUP.txt")
     params:
         threads = clusterconf["dedup"]["threads"],
         samplename = get_samplename,
         sentieon = pipeconfig["singularities"]["sentieon"]["tool_path"],
     singularity:
         pipeconfig["singularities"]["sentieon"]["sing"]
+    shadow:
+        pipeconfig["rules"].get("dedup", {}).get("shadow", pipeconfig.get("shadow", False))
     shell:
         "echo $HOSTNAME;"
-        "shellbamfiles=$(echo {input.bamfiles} | sed 's/ / -i /g') ;"
         "{params.sentieon} driver -t {params.threads} "
-            "-i $shellbamfiles "
+            "-i {input.bam} "
             "--algo LocusCollector "
             "--fun score_info "
-            "{wildcards.workingdir}/{wildcards.stype}/dedup/{wildcards.sname}_DEDUP_score.txt ;"
+            "{output.score}; "
         "{params.sentieon} driver "
             "-t {params.threads} "
-            "-i $shellbamfiles "
+            "-i {input.bam} "
             "--algo Dedup "
             "--rmdup "
-            "--score_info {wildcards.workingdir}/{wildcards.stype}/dedup/{wildcards.sname}_DEDUP_score.txt "
-            "--metrics {wildcards.workingdir}/{wildcards.stype}/dedup/{wildcards.sname}_DEDUP.txt "
-            "{wildcards.workingdir}/{wildcards.stype}/dedup/{wildcards.sname}_DEDUP.bam"
+            "--score_info {output.score} "
+            "--metrics {output.metrics} "
+            "{output.bam}"
 
 rule realign_mapping:
     input:
-        "{workingdir}/{stype}/dedup/{sname}_DEDUP.bam"
+        bam = "{stype}/dedup/{sname}_DEDUP.bam",
+        bai = "{stype}/dedup/{sname}_DEDUP.bam.bai"
     singularity:
         pipeconfig["singularities"]["sentieon"]["sing"]
     output:
-        "{workingdir}/{stype}/realign/{sname}_REALIGNED.bam"
+        bam = temp("{stype}/realign/{sname}_REALIGNED.bam"),
```

Comment:
the output from realign mapping can't be temp, they look at this bam file in igv